apps/cowswap-frontend/src/widgetEventEmitter.ts

```diff
@@ -1,5 +1,201 @@
+import { CowAnalytics } from '@cowprotocol/analytics'
 import { CowWidgetEventPayloadMap, CowWidgetEvents, SimpleCowEventEmitter } from '@cowprotocol/events'
 
+const getCowAnalytics = (): CowAnalytics | undefined => {
+  if (typeof window !== 'undefined' && 'cowAnalyticsInstance' in window) {
+    return window.cowAnalyticsInstance as CowAnalytics
+  }
+  return undefined
+}
+
 export const WIDGET_EVENT_EMITTER = Object.freeze(
   new SimpleCowEventEmitter<CowWidgetEventPayloadMap, CowWidgetEvents>(),
 )
+
+interface TrackingEventPayload {
+  eventCategory?: string
+  eventAction?: string
+  eventLabel?: string
+  [key: string]: string | number | boolean | undefined
+}
+
+interface OrderTrackingPayload extends TrackingEventPayload {
+  walletAddress: string
+  orderId: string
+  chainId: number
+  sellToken: string
+  buyToken: string
+  sellAmount: string
+  buyAmount: string
+}
+
+interface PostedOrderPayload {
+  owner: string
+  orderUid: string
+  chainId: number
+  inputToken?: { address?: string; symbol?: string }
+  outputToken?: { address?: string; symbol?: string }
+  inputAmount?: { toString(): string }
+  outputAmount?: { toString(): string }
+  orderType: string
+  kind: string
+  receiver?: string
+  orderCreationHash?: string
+}
+
+interface OrderPayload {
+  chainId: number
+  order?: {
+    owner?: string
+    uid?: string
+    sellToken?: string
+    buyToken?: string
+    sellAmount?: string
+    buyAmount?: string
+    executedSellAmount?: string
+    executedBuyAmount?: string
+    executedFeeAmount?: string
+  }
+  transactionHash?: string
+}
+
+// Type guard functions to check payload types
+function isPostedOrderPayload(payload: unknown): payload is PostedOrderPayload {
+  return !!(
+    payload &&
+    typeof payload === 'object' &&
+    'owner' in payload &&
+    'orderUid' in payload &&
+    'chainId' in payload
+  )
+}
+
+function isOrderPayload(payload: unknown): payload is OrderPayload {
+  return !!(
+    payload &&
+    typeof payload === 'object' &&
+    'chainId' in payload &&
+    ('order' in payload || 'transactionHash' in payload)
```

Comment:
In case of a transactionHash, that's a tx rather than an order, no?
Unless... that's an ethflow order?!

Reply:
Did some changes + inline comments where added now. Could you re-check?